src/Engine/Combat/CombatEngine.java

```diff
@@ -1,9 +1,21 @@
 package Engine.Combat;
 
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import Engine.Combat.StrikeParams.BattleParams;
 import Terrain.GameMap;
 import Terrain.MapMaster;
 import Units.Unit;
+import Units.WeaponModel;
 
+/**
+ * CombatEngine provides COs with two levels of control of what goes on in combat:
+ * changeCombatContext(), which takes in a CombatContext and allows the COs to fiddle with just about anything.
+ * applyCombatModifiers(), which is for directly modifying a unit's firepower in a given attack.
```

Comment:
These comments don't seem to be accurate anymore.

Reply:
Just the function names, I think, but good catch.